.github/workflows/release.yaml

```diff
@@ -1,88 +1,34 @@
-        required: true
-      kuadrantOperatorVersion:
-        description: Kuadrant Operator version
-        default: 0.0.0
-        type: string
-      authorinoOperatorVersion:
-        description: Authorino Operator bundle version
-        default: 0.0.0
-        type: string
-      limitadorOperatorVersion:
-        description: Limitador Operator bundle version
-        default: 0.0.0
-        type: string
-      dnsOperatorVersion:
-        description: DNS Operator bundle version
-        default: 0.0.0
-        type: string
-      wasmShimVersion:
-        description: WASM Shim version
-        default: 0.0.0
-        type: string
-      consolePluginImageURL:
-        description: ConsolePlugin image URL
-        default: "quay.io/kuadrant/console-plugin:latest"
-        type: string
-      prerelease:
-        description: Is the release a pre-release?
-        required: false
-        type: boolean
+  pull_request:
+    types:
+    - closed
+    branches:
+      - 'release-v[0-9]+.[0-9]+'
 
 jobs:
-  build:
+  if_merged:
+    if: github.event.pull_request.merged == true
     name: Release operator
     runs-on: ubuntu-latest
     steps:
+      # TODO: not sure if the `Install gettext-base`is still required
       - name: Install gettext-base
         run: |
           sudo apt-get update
           sudo apt-get install -y gettext-base
-      - name: Set up Go 1.21.x
-        uses: actions/setup-go@v4
-        with:
-          go-version: 1.21.x
-        id: go
       - name: Checkout code at git ref
         uses: actions/checkout@v4
-        with:
-          ref: ${{ inputs.gitRef }}
-          token: ${{ secrets.KUADRANT_DEV_PAT }}
-      - name: Create release branch
-        if: ${{ !startsWith(inputs.gitRef, 'release-v') }}
+      - name: Set envoironment variables
         run: |
-          git checkout -b release-v${{ inputs.kuadrantOperatorVersion }}
-      - name: Prepare release
-        run: |
-          make prepare-release \
-          VERSION=${{ inputs.kuadrantOperatorVersion }} \
-          AUTHORINO_OPERATOR_VERSION=${{ inputs.authorinoOperatorVersion }} \
-          LIMITADOR_OPERATOR_VERSION=${{ inputs.limitadorOperatorVersion }} \
-          DNS_OPERATOR_VERSION=${{ inputs.dnsOperatorVersion }} \
-          WASM_SHIM_VERSION=${{ inputs.wasmShimVersion }} \
-          RELATED_IMAGE_CONSOLEPLUGIN=${{ inputs.consolePluginImageURL }}
-      - name: Commit and push
-        uses: stefanzweifel/git-auto-commit-action@v5
-        with:
-          commit_message: "Prepared release v${{ inputs.kuadrantOperatorVersion }}"
-          commit_user_name: "github-actions[bot]"
-          commit_user_email: "github-actions[bot]@users.noreply.github.com"
-          branch: release-v${{ inputs.kuadrantOperatorVersion }}
-          create_branch: true
-          tagging_message: v${{ inputs.kuadrantOperatorVersion }}
-          commit_options: '--signoff'
+          bash ./hack/release/load_envvar.sh
```

Comment:
You're not going to be able to modify the caller's shell because it's in a different process context.

Reply:
You sure? I am almost sure I got that from GitHubs examples. Will check.

Reply:
Oh.. that is then different. It is populating github environment which is a data structure and it is not shell env var. Bear in mind that this approach does not work locally, only on github workflows. That can be limitting.

Reply:
Yes I hit that issue. The only step in that uses the data structure values is the `softprops/action-gh-release@v1` which is creating the github release. There is no other scripts using that data. 

And of local checking I have add a `--echo` flag that will echo out what would be added to the the structure. So I think we are okay.